pulsar-client-cpp/tests/ChunkMessageIdImplTest.cc

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <pulsar/MessageId.h>
+#include "lib/MessageIdUtil.h"
+#include "ChunkMessageIdImpl.h"
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+using namespace pulsar;
+
+TEST(ChunkMessageIdImplTest, testSerialization) {
+    MessageId msgId(MessageId(1, 1, 1, 1), MessageId(2, 2, 2, 2));
+
+    std::string serialized;
+    msgId.serialize(serialized);
+
+    MessageId deserialized = MessageId::deserialize(serialized);
+
+    ASSERT_EQ(msgId, deserialized);
+}
+
+TEST(ChunkMessageIdImplTest, testCompareLedgerAndEntryId) {
+    MessageId id1(MessageId(1, 1, 1, 1), MessageId(-1, 2L, 1L, 0));
+    MessageId id2(MessageId(1, 1, 1, 1), MessageId(-1, 2L, 1L, 1));
+    MessageId id3(MessageId(1, 1, 1, 1), MessageId(-1, 2L, 2L, 0));
+    MessageId id4(MessageId(1, 1, 1, 1), MessageId(-1, 3L, 0L, 0));
```

Comment:
Both using or not using the implicit conversion are okay. But it's better to make the constructor code style consistent. 

i.e. both

```c++
MessageId id1(1, 1, 1, 1);
MessageId id2(1, 2, 3, 4);
```

and

```
MessageId id1(1, 1L, 1L, 1);
MessageId id2(1, 2L, 3L, 4);
```

are okay. But

```c++
MessageId id1(1, 1, 1, 1);
MessageId id2(1, 2L, 3L, 4);
```

looks bad.